src/test/operators/join_semi_anti_test.cpp

```diff
@@ -83,4 +83,66 @@ TEST_F(JoinSemiAntiTest, AntiJoinBig) {
+                        JoinMode::Semi, "resources/test_data/tbl/int.tbl", 1, join_predicates);
+}
+
+TEST_F(JoinSemiAntiTest, MultiPredicateSemiJoinRefSegments) {
+  std::vector<JoinPredicate> join_predicates;
+  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+
+  auto scan_a = this->create_table_scan(_table_wrapper_k, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_a->execute();
+
+  auto scan_b = this->create_table_scan(_table_wrapper_a, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_b->execute();
+
+  test_hash_join_output(scan_a, scan_b, {ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals, JoinMode::Semi,
+                        "resources/test_data/tbl/int.tbl", 1, join_predicates);
+}
+
+TEST_F(JoinSemiAntiTest, MultiPredicateSemiJoinBig) {
+  std::vector<JoinPredicate> join_predicates;
+  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+
+  test_hash_join_output(_table_wrapper_semi_a, _table_wrapper_semi_b, {ColumnID{0}, ColumnID{0}},
+                        PredicateCondition::Equals, JoinMode::Semi,
+                        "resources/test_data/tbl/joinoperators/semi_result.tbl", 1, join_predicates);
+}
+
+TEST_F(JoinSemiAntiTest, MultiPredicateAntiJoin) {
+  std::vector<JoinPredicate> join_predicates;
+  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+
+  test_hash_join_output(_table_wrapper_k, _table_wrapper_a, {ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals,
+                        JoinMode::Anti, "resources/test_data/tbl/joinoperators/anti_int4.tbl", 1, join_predicates);
+}
+
+TEST_F(JoinSemiAntiTest, MultiPredicateAntiJoinRefSegments) {
+  std::vector<JoinPredicate> join_predicates;
+  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+
+  auto scan_a = this->create_table_scan(_table_wrapper_k, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_a->execute();
+
+  auto scan_b = this->create_table_scan(_table_wrapper_a, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_b->execute();
+
+  test_hash_join_output(scan_a, scan_b, {ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals, JoinMode::Anti,
+                        "resources/test_data/tbl/joinoperators/anti_int4.tbl", 1, join_predicates);
+}
+
+TEST_F(JoinSemiAntiTest, MultiPredicateAntiJoinBig) {
+  std::vector<JoinPredicate> join_predicates;
+  join_predicates.emplace_back(JoinPredicate{ColumnIDPair{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+
+  test_hash_join_output(_table_wrapper_semi_a, _table_wrapper_semi_b, {ColumnID{0}, ColumnID{0}},
+                        PredicateCondition::Equals, JoinMode::Anti,
+                        "resources/test_data/tbl/joinoperators/anti_result.tbl", 1, join_predicates);
+}
+
```

Comment:
We suggest to implement more diverse tests, i.e.,

- more than one additional join predicate
- an additional join predicate that uses different columns than the column already used by the normal join
- a join predicate not using columns 0 and 0
- join predicates with multiple non-ascending column ids, for example `1<>2, 2<>5`

To reduce boilerplate, you might want to look into using `INSTANTIATE_TEST_CASE_P` and `TEST_P`. Here's an example: https://github.com/hyrise/hyrise/blob/95c57d4f1c185018035a5bbde26530405a11492c/src/test/operators/table_scan_sorted_test.cpp#L138-L171

Reply:
Thank you for the suggestions.